src/handlers/backport.rs

```diff
@@ -0,0 +1,241 @@
+});
+
+const BACKPORT_LABELS: [&str; 4] = [
+    "beta-nominated",
+    "beta-accepted",
+    "stable-nominated",
+    "stable-accepted",
+];
+
+const REGRESSION_LABELS: [&str; 3] = [
+    "regression-from-stable-to-nightly",
+    "regression-from-stable-to-beta",
+    "regression-from-stable-to-stable",
+];
+
+// auto-nominate for backport only patches fixing high/critical regressions
+// For `P-{medium,low}` regressions, let the author decide
+const PRIORITY_LABELS: [&str; 2] = ["P-high", "P-critical"];
+
+#[derive(Default)]
+pub(crate) struct BackportInput {
+    // Issue(s) fixed by this PR
+    ids: Vec<u64>,
+    // Labels profile, compound value of (required_issue_label -> add_labels)
+    profile_labels: HashMap<String, Vec<String>>,
+}
+
+pub(super) async fn parse_input(
+    _ctx: &Context,
+    event: &IssuesEvent,
+    config: Option<&BackportTeamConfig>,
+) -> Result<Option<BackportInput>, String> {
+    let config = match config {
+        Some(config) => config,
+        None => return Ok(None),
+    };
+
+    // Only handle events when the PR is opened or the first comment is edited
+    if !matches!(event.action, IssuesAction::Opened | IssuesAction::Edited) && !event.issue.is_pr()
+    {
+        log::info!(
+            "Skipping backport event because: IssuesAction = {:?} issue.is_pr() {}",
+            event.action,
+            event.issue.is_pr()
+        );
+        return Ok(None);
+    }
+    let pr = &event.issue;
+
+    let pr_labels: Vec<&str> = pr.labels.iter().map(|l| l.name.as_str()).collect();
+    if contains_any(&pr_labels, &BACKPORT_LABELS) {
+        log::debug!("PR #{} already has a backport label", pr.number);
+        return Ok(None);
+    }
+
+    // Retrieve backport config for this PR, based on its team label(s)
+    // If the PR has no team label matching any [backport.*.required_pr_labels] config, the backport labelling will be skipped
+    let mut input = BackportInput::default();
+    let valid_configs: Vec<_> = config
+        .configs
+        .iter()
+        .clone()
+        .filter(|(_cfg_name, cfg)| {
+            let required_pr_labels: Vec<&str> =
+                cfg.required_pr_labels.iter().map(|l| l.as_str()).collect();
+            if !contains_any(&pr_labels, &required_pr_labels) {
+                log::warn!(
+                    "Skipping backport nomination: PR is missing one required team label: {:?}",
+                    pr_labels
+                );
+                return false;
+            }
+            input
+                .profile_labels
+                .insert(cfg.required_issue_label.clone(), cfg.add_labels.clone());
+            true
+        })
+        .collect();
+    if valid_configs.is_empty() {
+        log::warn!(
+            "Skipping backport nomination: could not find a suitable backport config. Please ensure the triagebot.toml has a `[backport.*.required_pr_labels]` section matching the team label(s) for PR #{}.",
+            pr.number
+        );
```

Comment:
Is this supposed to be a user-facing warning? Given how it's written, it feels like it.